ts/features/pn/aar/store/reducers/index.ts

```diff
@@ -0,0 +1,145 @@
+  displayingNotificationData: "displayingNotificationData",
+  notAddresseeFinal: "notAddresseeFinal",
+  ko: "ko"
+} as const;
+
+export const validAARStatusTransitions = new Map<
+  AARFlowState["type"],
+  Set<AARFlowState["type"]>
+>([
+  [sendAARFlowStates.none, new Set([sendAARFlowStates.displayingAARToS])],
+  [
+    sendAARFlowStates.displayingAARToS,
+    new Set([sendAARFlowStates.fetchingQRData])
+  ],
+  [
+    sendAARFlowStates.fetchingQRData,
+    new Set([sendAARFlowStates.fetchingNotificationData, sendAARFlowStates.ko])
+  ],
+  [
+    sendAARFlowStates.fetchingNotificationData,
+    new Set([
+      sendAARFlowStates.displayingNotificationData,
+      sendAARFlowStates.notAddresseeFinal,
+      sendAARFlowStates.ko
+    ])
+  ],
+  [
+    sendAARFlowStates.displayingNotificationData,
+    new Set([sendAARFlowStates.ko])
+  ],
+  [sendAARFlowStates.notAddresseeFinal, new Set([sendAARFlowStates.ko])],
+  [
+    sendAARFlowStates.ko,
+    new Set([
+      sendAARFlowStates.fetchingQRData,
+      sendAARFlowStates.fetchingNotificationData,
+      sendAARFlowStates.displayingNotificationData,
+      sendAARFlowStates.none
+    ])
+  ]
+]);
+
+type FlowStates = typeof sendAARFlowStates;
+
+type NotInitialized = {
+  type: FlowStates["none"];
+};
+
+type DisplayingTos = {
+  type: FlowStates["displayingAARToS"];
+  qrCode: string;
+};
+
+type FetchQR = {
+  type: FlowStates["fetchingQRData"];
+  qrCode: string;
+};
+
+type FetchNotification = {
+  type: FlowStates["fetchingNotificationData"];
+  iun: string;
+  fullNameDestinatario: string;
+  mandateId?: string;
+};
+
+type DisplayingNotification = {
+  type: FlowStates["displayingNotificationData"];
+  fullNameDestinatario: string;
+  notification: ThirdPartyMessage;
+  mandateId?: string;
+};
+
+type FinalNotAddressee = {
+  type: FlowStates["notAddresseeFinal"];
+  fullNameDestinatario: string;
+  qrCode: string;
+  iun: string;
+};
+
+type ErrorState = {
+  type: FlowStates["ko"];
+  errorKind?: string;
+  previousState: AARFlowState;
+};
+
+export type AARFlowState =
+  | NotInitialized
+  | DisplayingTos
+  | FetchQR
+  | FetchNotification
+  | DisplayingNotification
+  | FinalNotAddressee
+  | ErrorState;
+
+export const INITIAL_AAR_FLOW_STATE: AARFlowState = {
+  type: "none"
+};
+
+export const isValidAARStateTransition = (
+  currentState: AARFlowState,
+  nextState: AARFlowState
+): boolean => {
+  const currentType = currentState.type;
+  const nextType = nextState.type;
+
+  const allowedNextStates = validAARStatusTransitions.get(currentType);
+  return allowedNextStates?.has(nextType) ?? false;
+};
+
+/**
+ * A reducer to store the AAR flow state
+ */
+export const aarFlowReducer = (
+  state: AARFlowState = INITIAL_AAR_FLOW_STATE,
+  action: Action
+): AARFlowState => {
+  switch (action.type) {
+    case getType(setAarFlowState):
+      return isValidAARStateTransition(state, action.payload)
+        ? action.payload
+        : state;
+
```

Comment:
Theoretically we should never dispatch a state change action that points to an unexpected state, so wouldn't it be better to force an error state in case, since this in an unexpected behavior? 

I believe that falling back to the current state could also brick the flow in certain conditions, by keeping it in an infinite loop of 
```
"action gotoState1"->
"saga listens to action and runs" ->
 "saga dispatches invalid action, back to state 1" -> 
 loop starts over, with no exit condition. 
```
what do you think?